code/game/machinery/poolcontroller.dm

```diff
@@ -0,0 +1,121 @@
+/obj/machinery/poolcontroller
+	name = "Pool Controller"
+	desc = "A controller for the nearby pool."
+	icon = 'icons/obj/airlock_machines.dmi'
+	icon_state = "airlock_control_standby"
+	var/list/linkedturfs = list() //List contains all of the linked pool turfs to this controller, assignment happens on New()
+	var/temperature = "normal" //The temperature of the pool, starts off on normal, which has no effects.
+	var/srange = 5 //The range of the search for pool turfs, change this for bigger or smaller pools.
+
+/obj/machinery/poolcontroller/New() //This proc automatically happens on world start
+	for(var/turf/simulated/floor/beach/water/W in range(srange,src)) //Search for /turf/simulated/floor/beach/water in the range of var/srange
+		src.linkedturfs += W //Add found pool turfs to the central list.
+	processing_objects.Add(src) //Add the pool controller to the processing queue to tie mob effects to the air controller.
```

Comment:
Use ..() instead of processing_objects.Add(). It should get added automatically.